clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -87,8 +92,8 @@ public void run() {
                 try {
                     runOnce();
                 } catch (final Throwable e) {
-                    // Swallow the exception and continue
```

Comment:
This seems related to the jira comment in https://issues.apache.org/jira/projects/KAFKA/issues/KAFKA-15867.

@philipnee @kirktrue do you folks have more context?

Reply:
@jeffkbkim I have the answer for that question already, which is yes - we should tell the user what had gone wrong in the background thread via propagating the exception back to polling thread

Reply:
I understand, but I am wondering why we added
```
// Swallow the exception and continue
```
in the first place. 

The jira also asks:
> may you give some insights on why we don't want to propagate the error to the caller? 

@kirktrue can you answer this?

Reply:
Yes, we should communicate these errors to the application thread.